### DISCUSSION TITLE
Color substring of text among Textbox

### DISCUSSION THREAD

**Discussion by MoritzNekolla** [TomSchimansky/CustomTkinter#1603] [General]
Hello,

is there a possibility to color just a substring of the text inside a `customtinker.Textbox`? Or highlight it otherwise?

In plain tkinter this is possible like so:

```python
import tkinter as tk

# Create a new tkinter window
root = tk.Tk()

# Create a Text widget
text_widget = tk.Text(root, height=10, width=50)
text_widget.pack()

# Add text to the Text widget
text_widget.insert('end', 'This is some colored text.\n')

# Add a tag to color a substring
text_widget.tag_add('color', '1.5', '1.11') # '1.5' means first line, 5th character; '1.11' means first line, 11th character

# Configure the tag to set the text color
text_widget.tag_configure('color', foreground='red')

# Run the tkinter event loop
root.mainloop()
```

However customtinker only provides the `text_color` parameter, which colors the whole Text.

Any ideas?

**Comment by TomSchimansky** [comment]
Works fine, but the method is called `tag_config` as in the tkinter docs. It's really confusing in tkinter that there exist multiple names for the same method. I only wrapped `tag_config` for CustomTkinter.

```python
import customtkinter

app = customtkinter.CTk()

text_widget = customtkinter.CTkTextbox(app)
text_widget.pack()

text_widget.insert('end', 'This is some colored text.\n')
text_widget.tag_add('color', '1.5', '1.11')  # '1.5' means first line, 5th character; '1.11' means first line, 11th character
text_widget.tag_config('color', foreground='red')

app.mainloop()
```

**Reply by UltronXx** [reply]
How do you get first line, 10th character?

**Reply by MoritzNekolla** [reply]
text_widget.tag_add('color', '1.10', '1.11') i guess

**Reply by UltronXx** [reply]
Yes... I didn't know it was a string argument
Thank you!!!!

**Comment by MoritzNekolla** [comment]
@TomSchimansky 

Could you explain how I color substrings of a `CTkLabel` as well? `tag_add()` does not work here neither does Latex Strings.

Thanks in advance

**Comment by Wizard0f0zz** [comment]
Hi,
I have written a code to append text in the textbox with different styles.
```
import customtkinter as ctk
global textbox

def insert_styled_text(textbox, text, color="black"):
    match color:
        case 'black':
            tag_name = 'black'
            textbox.tag_config(tag_name, foreground=color)
        case 'blue':
            tag_name = 'blue'
            textbox.tag_config(tag_name, foreground=color, overstrike=True)
        case 'red':
            tag_name = 'red'
            textbox.tag_config(tag_name, foreground=color, underline=True)

    textbox.insert("end", text, tag_name)
    textbox.yview_moveto(1)

def create_app():
    root = ctk.CTk()
    textbox = ctk.CTkTextbox(root, width=400, height=200, fg_color='white')
    textbox.pack(padx=10, pady=10)

    # Insert text with styles

    insert_styled_text(textbox, "Hello, World! ", color="blue", )
    insert_styled_text(textbox, "This is custom text.", color="red")
    insert_styled_text(textbox, "This is some more custom text.\n")
    root.mainloop()

# Run the App
create_app()
```

I can create multiple tags to generate different styles, but unable to add bold and italics as tag_config does not support it. 
Alternatively I tried to add it to the font in defining the CTkTextbox, but it bolds all the text in the textbox.
underline and overstrike works as shown in the code above. Is there a way I can partially make the text bold or in italics?

Any help will be greatly appreciated.
Thank you.
Cheers!!

**Reply by dipeshSam** [reply]
@Wizard0f0zz Here is the solution: https://github.com/TomSchimansky/CustomTkinter/discussions/2463#discussioncomment-9733801

Regards.

**Reply by Wizard0f0zz** [reply]
Thank you so much Dipesh. It worked perfectly.

**Reply by dipeshSam** [reply]
@Wizard0f0zz You are most welcome.
